src/components/profile/EditProfileModal.jsx

```diff
@@ -0,0 +1,199 @@
+import React, { Component } from 'react';
+import Modal from 'react-modal';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+
+import './EditProfileModal.scss';
+import uploadImage from '../../utils/imageUpload';
+import { updateProfile } from '../../actions/profile';
+
+Modal.setAppElement(document.querySelector('#root'));
+
+/**
+ * @class EditProfileModal
+ * @description renders a modal on the profile page
+ * @returns {JSX} - rendered output
+ */
+class EditProfileModal extends Component {
+  /**
+   * @constructor
+   * @param {*} props react props
+   * @memberof EditProfileModal
+   * @returns {void}
+   */
+  constructor(props) {
+    super(props);
+    this.fileInput = React.createRef();
+  }
+
+  state = {
+    username: '',
+    bio: '',
+    email: '',
+    image: '',
+    uploading: false,
+  };
+
+  /**
+   * @method - componentWillReceiveProps
+   * @description - a React lifecylce method
+   * @returns {void}
+   */
+  componentWillReceiveProps() {
```

Comment:
Hey @frostyblok, for the current use case componentDidUpdate wouldn't work because I'm currently setting state within the lifecycle method so this would lead to an infinite loop.

I'll try exploring other options though.

Reply:
This has been fixed 🎉

Reply:
Good job, Kizito! This is good to go.